test/unit/tasks/usage_limits_test.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class UsageLimitsTest < ActiveSupport::TestCase
+  include ActiveJob::TestHelper
+
+  test 'executing the task removes orphaned usage limits' do
+    service = FactoryBot.create(:simple_service)
+    service_metric = service.metrics.first
+    plan = FactoryBot.create(:application_plan_without_rules, issuer: service)
+
+    backend_api1 = FactoryBot.create(:backend_api, account: service.account)
+    backend_metric1 = FactoryBot.create(:metric, owner: backend_api1)
+    backend_usage_limit1 = FactoryBot.create(:usage_limit, metric: backend_metric1, plan: plan)
```

Comment:
Actually, I'm not sure we should even allow creating this usage limit, given that the backend and service are not linked :thinking: 

Probably though, on the controller level it will be still restricted, but not sure we need to also restrict it at a model level?

WDYT @akostadinov @jlledom ?

Reply:
Then we would need to handle a situation in which backend is removed from a service... if we have such a metric, it will just never be updated or what? Are there any negative side effects? If no side effects, then I wouldn't add more logic to it.

Reply:
> Then we would need to handle a situation in which backend is removed from a service... if we have such a metric, it will just never be updated or what?

Well, this whole PR is for fixing this situation :grimacing: 

What it does is - when a backend usage is removed from a service, all of the usage limits (on all service's plans) associated to the metrics of this backend are deleted.

Also, there is a rake task that removes all usage limits that belong to backends that are not associated to the service (which plan has these limits).

Reply:
Then it makes sense to me to add a validation.

Reply:
But this test is precisely about `backend_usage_limit1` existing and then being removed by the rake task. If you don't create `backend_usage_limit1`, how do you check what the rake task does? I'm missing something...